src/main/java/net/dv8tion/jda/api/entities/MessageType.java

```diff
@@ -148,25 +148,31 @@
      *
      * Messages from this type usually come with custom embeds containing relevant information, the author is the user that triggered the filter.
      */
-    AUTO_MODERATION_ACTION(24),
+    AUTO_MODERATION_ACTION(24, true, true),
 
     /**
      * Unknown MessageType.
      */
-    UNKNOWN(-1);
+    UNKNOWN(-1, false, true);
 
     private final int id;
     private final boolean system;
+    private final boolean deletable;
 
-    MessageType(int id)
-    {
-        this(id, true); // true as default because the majority are system
-    }
-
-    MessageType(int id, boolean system)
-    {
+    MessageType(int id, boolean system, boolean deletable) {
         this.id = id;
         this.system = system;
+        this.deletable = deletable;
+    }
+
+    MessageType(int id, boolean system) {
+        // most system messages are not deletable, most non-system messages are
+        this(id, system, !system);
+    }
+
+    MessageType(int id) {
```

Comment:
Remove these overload constructors and only have one with both booleans. I don't like having this implicit true and false here, its confusing.